### PR TITLE
fix(marketplace): `AppsInfo` small UX problems

### PR DESCRIPTION
**Pull request by tiagoevanp**
## Proposed changes (including videos or screenshots)

- Fix the space between tags:
image

- Fix webpage and support links parser

## Further comments

AECO-304
AECO-239

### CODE REVIEW

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppDetails/AppDetails.tsx [hunk 1]
```diff
@@ -9,6 +10,18 @@ import ScreenshotCarouselAnchor from '../../../components/ScreenshotCarouselAnch
...
+const normalizeUrl = (url: string): string => {
+	const parsedUrl = parse(url);
+
+	if (parsedUrl[0].type === 'PARAGRAPH') {
+		if (parsedUrl[0].value[0].type === 'LINK') {
+			return parsedUrl[0].value[0].value.src.value;
+		}
+	}
+
+	return url;
```
Comment (tassoevan) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppDetails/AppDetails.tsx:22]: Prefer early returns.

```suggestion
	const parsedUrl = parse(url);

	if (parsedUrl[0].type !== 'PARAGRAPH') {
		return url;
	}
	
	if (parsedUrl[0].value[0].type !== 'LINK') {
		return url;
	}
	
	return parsedUrl[0].value[0].value.src.value;
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppDetails/AppDetails.tsx [hunk 2]
```diff
@@ -9,6 +10,18 @@ import ScreenshotCarouselAnchor from '../../../components/ScreenshotCarouselAnch
 import type { AppInfo } from '../../../definitions/AppInfo';
 import AppDetailsAPIs from './AppDetailsAPIs';
 
+const normalizeUrl = (url: string): string => {
```
Comment (tassoevan) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppDetails/AppDetails.tsx:13]: What if `url` is not a valid linkable text?

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx [hunk 3]
```diff
@@ -146,7 +146,7 @@ const AppStatus = ({ app, showStatus = true, isAppDetailsPage, installed, ...pro
 			)}
 
 			{statuses?.map((status, index) => (
-				<Margins inlineEnd='x8' key={index}>
+				<Margins inline='4px' key={index}>
```
Comment (tassoevan) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx:149]:
```suggestion
				<Margins inline={4} key={index}>
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatusPriceDisplay.tsx [hunk 4]
```diff
@@ -24,10 +24,12 @@ const AppStatusPriceDisplay: FC<AppStatusPriceDisplayProps> = ({ purchaseType, p
 	);
 
 	return (
-		<Tag>
-			{showType && <Box color='default'>{t(type as TranslationKey)}</Box>}
-			<Box>{!showType && type === 'Free' ? t(type) : formattedPrice}</Box>
-		</Tag>
+		<Margins inline='4px'>
```
Comment (tassoevan) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatusPriceDisplay.tsx:27]:
```suggestion
		<Margins inline={4}>
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatusPriceDisplay.tsx [hunk 5]
```diff
@@ -24,10 +24,12 @@ const AppStatusPriceDisplay: FC<AppStatusPriceDisplayProps> = ({ purchaseType, p
...
+			<Tag>
+				{showType && <Box color='default'>{t(type as TranslationKey)}</Box>}
```
Comment (tassoevan) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatusPriceDisplay.tsx:29]: What if `type` is not a translation key? Why not type-guard it?

**Review by tassoevan — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppDetails/AppDetails.tsx (hunk 2 above)
Comment (tiagoevanp) [reply apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppDetails/AppDetails.tsx:13]: return a empty string...

Comment (tassoevan) [reply apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppDetails/AppDetails.tsx:13]: These lines contradict it: https://github.com/RocketChat/Rocket.Chat/pull/30012/files#diff-46795d9738beab48b68302a1312498cf6e19a77d2efd1a6981f5155e5e632716R20-R26

**Review by tassoevan — CHANGES_REQUESTED** [review]

**Review by tassoevan — APPROVED** [review]

**Pull request merged**